### PR TITLE
Reverse circular linked list

### PR DESCRIPTION
**Pull request by khusheekapoor**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the Contributing Guidelines on Pull Requests

### Description

To reverse a circularly linked list in Java as a GSSoC'21 participant.

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #5748

### CODE REVIEW

File: Java/README.md [hunk 1]
```diff
@@ -170,6 +170,7 @@ _add list here_
 - [Double Ended Queue in Java](ds/DEQueue.java)
 - [Circular Queue](ds/CircularQueue.java)
 - [Implementing Stacks using Arrays](ds/StackUsingArrays.java)
+- [Reversing Ciruclar Linked List](ds/ReversingCircularLinkedList.java)
```
Comment (ankitaggarwal23) [inline Java/README.md]: Fix typo

File: Java/ds/ReversingCircularLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+ Enter data in Circular LinkedList: 
+ 6
+ Enter data in Circular LinkedList: 
+ 7
+ Enter data in Circular LinkedList: 
+ 8
+ Original Circular LinkedList: 
+ 6 7 8 
+ Reversed Circular LinkedList: 
+ 8 7 6
```
Comment (ankitaggarwal23) [inline Java/ds/ReversingCircularLinkedList.java]: Remove extra test case

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Java/README.md (hunk 1 above)
Comment (khusheekapoor) [reply Java/README.md]: Made the changes.

File: Java/ds/ReversingCircularLinkedList.java (hunk 2 above)
Comment (khusheekapoor) [reply Java/ds/ReversingCircularLinkedList.java]: Made the changes.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**